king_phisher/client/windows/plugin_manager.py

```diff
@@ -135,6 +168,76 @@ def load_plugins(self):
 				"{0} (Load Failed)".format(name)
 			))
 
+		if not self.catalog_plugins.catalog_ids():
+			return
+		if not self.catalog_plugins.get_cache():
+			return
+
+		if self.catalog_plugins.catalog_ids():
+			for catalogs in self.catalog_plugins.catalog_ids():
+				catalog = store.append(None, (catalogs, True, None, 'Catalog', catalogs, None, None, False, False))
+				for repos in self.catalog_plugins.get_repos(catalogs):
+					repo = store.append(catalog, (repos.id, True, None, 'Repository', repos.title, None, None, False, False))
+					plugin_collections = self.catalog_plugins.get_collection(catalogs, repos.id)
+					client_plugins = list(plugin_collections)
+					client_plugins.sort()
+					for plugins in client_plugins:
+						installed = False
+						enabled = False
+						if plugin_collections[plugins]['name'] in self.config['plugins.installed']:
+							if repos.id == self.config['plugins.installed'][plugin_collections[plugins]['name']][1]:
+								installed = True
+								enabled = True if plugin_collections[plugins]['name'] in self.config['plugins.enabled'] else False
+						store.append(repo, (
+							plugins,
+							enabled,
+							installed,
+							'Plugin',
+							plugin_collections[plugins]['title'],
+							'Yes' if self.catalog_plugins.is_compatible(catalogs, repos.id, plugins) else 'No',
+							self.compare_plugin_versions(plugin_collections[plugins]['name'], plugin_collections[plugins]['version']),
+							True,
+							True
+						))
+		else:
+			for catalog_id in self.catalog_plugins.get_cache_catalog_ids():
+				catalog_line = store.append(None, (catalog_id, True, None, 'Catalog (offline)', catalog_id, None, None, False, False))
+				for repo in self.catalog_plugins.get_cache_repos(catalog_id):
+					repo_line = store.append(catalog_line, (repo.id, True, None, 'Repository (offline)', repo.title, None, None, False, False))
+					for plugin in self.config['plugins.installed']:
+						if self.config['plugins.installed'][plugin][0] != catalog_id:
+							continue
+						if self.config['plugins.installed'][plugin][1] != repo.id:
+							continue
+						store.append(repo_line, (
+							plugin,
+							True if plugin in self.config['plugins.enabled'] else False,
+							True,
+							'Plugin',
+							pm[plugin].title,
+							'Yes' if pm[plugin].is_compatible else 'No',
+							pm[plugin].version,
+							True,
+							False
+						))
+
+	def compare_plugin_versions(self, plugin_name, plugin_version):
+		if plugin_name not in self.application.plugin_manager:
```

Comment:
I don't understand what this function is doing. The return type is inconsistent as is it's meaning, sometimes its a plugin, other times its "Upgrade available"? Seems like it should be renamed `is_upgrade_available` and return a boolean, the caller can then decide to translate that into a string.